site/src/components/code-block.tsx

```diff
@@ -192,8 +194,46 @@ const CodeBlock: React.FC<CodeBlockProps> = (props) => {
   );
 };
 
+/**
+ * Returns the number of spaces the provided string starts with.
+ */
+function numSpaces(str: string, spacesPerTab: number) {
+  let ret = 0;
+  for (let i = 0; i < str.length; i += 1) {
+    if (str.charAt(i) === ' ') {
+      ret += 1;
+    } else if (str.charAt(i) === '\t') {
+      ret += spacesPerTab;
+    } else {
+      break;
+    }
+  }
+  return ret;
+}
+
+/**
+ * Calculates all existing indentations for each line, and normalizes the indentation.
+ * This is done by lining up the indentations and assigning a multiple of the provided {@param indent}.
+ * If a tab is found, then the {@param spacesPerTab} is used to convert to spaces.
+ */
+function normalizeIndentation(
+  code: string[],
+  indent: number = 2,
+  spacesPerTab: number = 4,
+): string[] {
```

Comment:
If I understood correctly, this is not capable of handling parameters aligned against an operning parenthesis, right? e.g.

```
foobar(foo,
       bar);

=>
foobar(foo,
  bar);
```